### PR TITLE
[JENKINS-74095] [JENKINS-74096] [JENKINS-74097] [JENKINS-74099] Remove legacy checkUrl handlers

### PR DESCRIPTION
**Pull request by shlomomdahan**
Jira: 
- [JENKINS-74095](https://issues.jenkins.io/browse/JENKINS-74095)
- [JENKINS-74096](https://issues.jenkins.io/browse/JENKINS-74096)
- [JENKINS-74097](https://issues.jenkins.io/browse/JENKINS-74097)
- [JENKINS-74099](https://issues.jenkins.io/browse/JENKINS-74099)

### What has been done
1. Remove the legacy onClick handlers as part of effort to improve CSP compatibility across Jenkins

Freestyle Project:
[Before Changes](https://www.loom.com/share/3c67b55aed4c47b7bb8eb054204cca66)
[After Changes](https://www.loom.com/share/288f1f2ca9684f4babeee6c23e374f97)

Pipeline Project:
[After Changes](https://www.loom.com/share/15aba7478b3e4604868f34fcccf68359)

### How to test - Freestyle Project
1. Create a freestyle project
3. In post build steps, select: Plot build data
4. Add the data series file and check validation triggers. 
Screenshot 2024-11-19 at 11 57 50 AM

### How to test - Pipeline Project
1. Create Pipeline Project
2. In configuration select Pipeline Syntax and navigate to snipper generator.
3. In Sample Step selector, select: `Plot: Plot build data`
4. add CSV, Properties, and XML Series data. 
5. Change the Series file name and verify in network tab that correct requests are sent. 

### Checklist

- [x] Git commits follow [best practices](https://chris.beams.io/posts/git-commit/) 
- [x] Build passes in Jenkins 
- [x] Appropriate tests or explanation to why this change has no tests 
- [ ] Pull Request is marked with appropriate label (see `.github/release-drafter.yml`) 
- [x] JIRA issue is well described (problem explanation, steps to reproduce, screenshots) 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### CODE REVIEW

**Comment by yaroslavafenkin** [comment]
> Remove legacy inline onClick handlers

Do you mean `checkUrl`s, not `onClick`?

File: src/main/resources/hudson/plugins/plot/CSVSeries/config.jelly [hunk 1]
```diff
@@ -3,9 +3,8 @@
          xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     <f:entry>
         <div>
-            <f:entry title="${%Data series file}" field="file" help="/plugin/plot/help-series.html">
-                <f:textbox name="file" value="${series.file}"
-                           checkUrl="'descriptorByName/hudson.plugins.plot.PlotPublisher/checkSeriesFile?value='+toValue(this)"/>
+            <f:entry title="${%Data series file}" field="seriesFile" help="/plugin/plot/help-series.html">
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/plot/CSVSeries/config.jelly:6]: Have you tested all of these occurrences? These do not always get automagically wired to the descriptor methods as I've recently discovered.

Comment (shlomomdahan) [reply src/main/resources/hudson/plugins/plot/CSVSeries/config.jelly]: Thanks for pointing this out. I assumed those were working the same way, but it turns out this is needed to compatibility with Pipeline #32 

I reverted back t the checkDependsOn approach

**Review by yaroslavafenkin — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
Thanks!

**Pull request merged**